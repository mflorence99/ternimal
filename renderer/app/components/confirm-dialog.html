<h3 mat-dialog-title>
  {{ title }}
</h3>

<p mat-dialog-content>
  {{ message }}
</p>

<div mat-dialog-actions>
  <button (click)="dismiss()" mat-button>
    No
  </button>

  <button
    [mat-dialog-close]="true"
    cdkFocusInitial
    color="primary"
    mat-raised-button
  >
    Yes
  </button>
</div>
