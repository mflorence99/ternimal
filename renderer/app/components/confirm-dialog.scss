:host {
  display: block;
  width: 20rem;
}
