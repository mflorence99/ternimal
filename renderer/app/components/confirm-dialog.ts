import { ChangeDetectionStrategy } from '@angular/core';
import { Component } from '@angular/core';
import { Inject } from '@angular/core';
import { MAT_DIALOG_DATA } from '@angular/material/dialog';
import { MatDialogRef } from '@angular/material/dialog';

export class ConfirmDialogModel {
  constructor(public title: string, public message: string) {}
}

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'ternimal-confirm-dialog',
  templateUrl: 'confirm-dialog.html',
  styleUrls: ['confirm-dialog.scss']
})
export class ConfirmDialogComponent {
  message: string;
  title: string;

  constructor(
    public dialogRef: MatDialogRef<ConfirmDialogComponent>,
    @Inject(MAT_DIALOG_DATA) public data: ConfirmDialogModel
  ) {
    this.message = data.message;
    this.title = data.title;
  }

  confirm(): void {
    this.dialogRef.close(true);
  }

  dismiss(): void {
    this.dialogRef.close(false);
  }
}
