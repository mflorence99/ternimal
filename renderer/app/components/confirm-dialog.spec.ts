import { ConfirmDialogComponent } from './confirm-dialog';
import { ConfirmDialogModel } from './confirm-dialog';

import { prepare } from './component.spec';

import 'jest-extended';

import { ComponentFixture } from '@angular/core/testing';
import { TestBed } from '@angular/core/testing';

describe('ConfirmDialogComponent', () => {
  let component: ConfirmDialogComponent;
  let fixture: ComponentFixture<ConfirmDialogComponent>;

  beforeEach(() => {
    prepare();
    fixture = TestBed.createComponent(ConfirmDialogComponent);
    component = fixture.componentInstance;
  });

  test('model', () => {
    const model = new ConfirmDialogModel('t', 'm');
    expect(model.title).toBe('t');
    expect(model.message).toBe('m');
  });

  test('confirm', () => {
    component.confirm();
    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(component.dialogRef.close).toHaveBeenCalledWith(true);
  });

  test('dismiss', () => {
    component.dismiss();
    // eslint-disable-next-line @typescript-eslint/unbound-method
    expect(component.dialogRef.close).toHaveBeenCalledWith(false);
  });

  test('snapshot', () => {
    component.message = 'A message';
    component.title = 'A title';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
