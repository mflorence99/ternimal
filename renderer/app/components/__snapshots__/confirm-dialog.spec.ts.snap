// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogComponent snapshot 1`] = `
<ternimal-confirm-dialog
  __ngContext__={[Function LRootView]}
  data={[Function Object]}
  dialogRef={[Function MockMatDialogRef]}
  message={[Function String]}
  title={[Function String]}
>
  <h3
    class="mat-dialog-title"
    mat-dialog-title=""
  >
    A title
  </h3><p
    class="mat-dialog-content"
    mat-dialog-content=""
  >
    A message
  </p><article
    class="mat-dialog-actions"
    mat-dialog-actions=""
  >
    <button
      class="mat-focus-indicator mat-button mat-button-base"
      mat-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        No
      </span>
      <span
        class="mat-ripple mat-button-ripple"
        matripple=""
      />
      <span
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      cdkfocusinitial=""
      class="mat-focus-indicator mat-raised-button mat-button-base mat-primary"
      color="primary"
      mat-raised-button=""
      type="button"
    >
      <span
        class="mat-button-wrapper"
      >
         Yes 
      </span>
      <span
        class="mat-ripple mat-button-ripple"
        matripple=""
      />
      <span
        class="mat-button-focus-overlay"
      />
    </button>
  </article>
</ternimal-confirm-dialog>
`;
